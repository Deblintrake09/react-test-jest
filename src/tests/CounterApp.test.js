import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe('Pruebas del componente <CounterApp/>', ()=>{

    let wrapper = shallow(<CounterApp/>);
    beforeEach(()=>{
        wrapper = shallow(<CounterApp/>);
    })
    
    test('debe mostrar el componente de manera correcta',()=>{
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe mostrar un valor por defecto de 100 en el value', ()=>{
        const wrapper = shallow(<CounterApp value={100}/>)
        const valueTest =wrapper.find('p').text().trim();
        expect(valueTest).toBe("100");
    })

    test('debe de incrementar con boton +1', ()=>{
        wrapper.find('button').at(0).simulate('click');
        const valueText =wrapper.find('p').text().trim();

        expect(valueText).toBe('11');
    })

    test('debe de disminuir el valor del contador con boton -1', ()=>{
        wrapper.find('button').at(2).simulate('click');
        const valueText =wrapper.find('p').text().trim();

        expect(valueText).toBe('9');
    })

    test('debe colocar el valor por defecto tras presionar el boton reset', ()=>{
        const wrapper = shallow(<CounterApp value={105}/>)
        
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        
        const valueText =wrapper.find('p').text().trim();
        
        expect(valueText).toBe('105');

    })
});