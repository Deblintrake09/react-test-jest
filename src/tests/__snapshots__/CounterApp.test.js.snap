// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <CounterApp/> debe mostrar el componente de manera correcta 1`] = `
<Fragment>
  <h1>
    Counter-App
  </h1>
  <p>
    10
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
